src/mlpack/bindings/go/CMakeLists.txt

```diff
@@ -24,13 +24,15 @@ if (BUILD_GO_BINDINGS)
     find_package(Gonum)
     if (NOT GO_FOUND OR NOT GONUM_FOUND)
       unset(BUILD_GO_BINDINGS CACHE)
+      set(BUILD_GO_SHLIB OFF)
       message(FATAL_ERROR "Go or Gonum not found; unable to build Go bindings!")
     endif()
   else ()
     find_package(Go 1.11.0)
     find_package(Gonum)
     if (NOT GO_FOUND OR NOT GONUM_FOUND)
       unset(BUILD_GO_BINDINGS CACHE)
+      set(BUILD_GO_SHLIB OFF)
```

Comment:
@Yashwants19 nice catch!  Do you think it would be better to instead call `not_found_return()` here?  Then hopefully none of the go-related targets even get added in CMake.

Reply:
If we use `not_found_return()` here, we are still building all go-shared library, because `BUILD_GO_SHLIB=ON` and hence we are adding all the go related targets.  BTW we are calling `not_found_return()` [here](https://github.com/mlpack/mlpack/blob/60ac0bea5cc100edbf8f8be7dd048449c9c66ff5/src/mlpack/bindings/go/CMakeLists.txt#L40), if cmake doesn't found any of the dependencies.

Reply:
Ok, sorry, maybe I was getting a little confused.  Thanks for the clarifications.  But I still haven't figured out why this is needed---`BUILD_GO_SHLIB` is only ever referenced from `src/mlpack/bindings/go/CMakeLists.txt`, so if we call `not_found_return()`, then it won't ever be used anyway since `not_found_return()` will stop any further processing of `src/mlpack/bindings/go/CMakeLists.txt`.  So I don't think I understand why we need to set it to `OFF`?

Reply:
I see, I did a bit of digging.  It turns out one of my assumptions is wrong:  I had thought that `not_found_return()` would exit the current CMake file.  But it turns out that this is not true!  I wonder if `not_found_return()` actually has to be a macro.  In any case, it's not a big deal, the changes you made seem to work fine.

Whatever you want to do here is fine with me, it works either way.  I think that the discussion here covers the other comment too. :+1:

Reply:
I had use `BUILD_GO_SHLIB` because when we ask the user to download go-bindings, then we also ask the user to build these go-shared libraries on their own, and this option will help the user to just build the shared libraries required by the go-bindings and not the complete go-bindings (.go and .h files).

Reply:
That's fair, all I mean is, if `not_found_return()` actually *did* stop processing `src/mlpack/bindings/go/CMakeLists.txt`, then we wouldn't need to set `BUILD_GO_SHLIB` like you did in this change.  I think the change you made is just fine, but another option could be to "fix" `not_found_return()` so it actually stops processing the file, and then these changes wouldn't be needed.

Reply:
Currently `not_found_return()` depends on both `BUILD_GO_BINDNIGS` and  `BUILD_GO_SHLIB` for stop processing the file.
It seems that `not_found_return()` must not depend only on `BUILD_GO_BINDNIGS`. If you don't mind, then I can open a PR for the "fix" afterwards.

Reply:
Either way is fine---the change you've made here seems to work.  In fact this is already approved so I'll go ahead and merge this now, and if you want to provide another update later feel free.